Rubberduck.VBEEditor/Class1.cs

```diff
@@ -0,0 +1,964 @@
+        public void /* ITypeInfo:: */ ReleaseTypeAttr(IntPtr pTypeAttr)
+            => _wrappedObject.ReleaseTypeAttr(pTypeAttr);
+        public void /* ITypeInfo:: */ ReleaseFuncDesc(IntPtr pFuncDesc)
+            => _wrappedObject.ReleaseFuncDesc(pFuncDesc);
+        public void /* ITypeInfo:: */ ReleaseVarDesc(IntPtr pVarDesc)
+            => _wrappedObject.ReleaseVarDesc(pVarDesc);
+
+        public IDispatch GetStdModInstance() => _ITypeInfoAlt.GetStdModInstance();
+        public object StdModExecute(string name, Reflection.BindingFlags invokeAttr, object[] args = null)
+        {
+            var StaticModule = GetStdModInstance();
+            var retVal = StaticModule.GetType().InvokeMember(name, invokeAttr, null, StaticModule, args);
+            Marshal.ReleaseComObject(StaticModule);
+            return retVal;
+        }
+
+        public TypeInfoWrapper_VBE GetImplementedTypeInfoByIndex(int implIndex)
+        {
+            ITypeInfo typeInfoImpl = null;
+            int href = 0;
+            GetRefTypeOfImplType(implIndex, out href);
+            GetRefTypeInfo(href, out typeInfoImpl);
+            return (TypeInfoWrapper_VBE)typeInfoImpl;
+        }
+
+        public bool DoesImplement(string interfaceName)
+        {
+            if (Name == interfaceName) return true;
+
+            for (int implIndex = 0; implIndex < _cachedAttributes.cImplTypes; implIndex++)
+            {
+                using (var typeInfoImplEx = GetImplementedTypeInfoByIndex(implIndex))
+                {
+                    if (typeInfoImplEx.DoesImplement(interfaceName))
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        public bool DoesImplement(Guid interfaceIID)
+        {
+            if (_cachedAttributes.guid == interfaceIID) return true;
+
+            for (int implIndex = 0; implIndex < _cachedAttributes.cImplTypes; implIndex++)
+            {
+                using (var typeInfoImplEx = GetImplementedTypeInfoByIndex(implIndex))
+                {
+                    if (typeInfoImplEx.DoesImplement(interfaceIID))
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        public TypeInfoWrapper_VBE GetImplementedTypeInfo(string searchTypeName)
+        {
+            for (int implIndex = 0; implIndex < _cachedAttributes.cImplTypes; implIndex++)
+            {
+                var typeInfoImplEx = GetImplementedTypeInfoByIndex(implIndex);
+                if (typeInfoImplEx.Name == searchTypeName)
+                {
+                    return typeInfoImplEx;
+                }
+                typeInfoImplEx.Dispose();
+            }
+
+            throw new ArgumentException($"TypeLibWrapper_VBE::GetImplementedTypeInfo failed. '{searchTypeName}' module not found.");
```

Comment:
I haven't checked usage yet, but this seems like a good place to `return null;` after logging a warning.

Reply:
I disagree here, since we have DoesImplement() to check if it does implement the interface, and then GetImplementedTypeInfo() is considered absolute